tests/components/mqtt/test_common.py

```diff
@@ -697,16 +735,21 @@ async def help_test_discovery_removal(
     assert state is None
 
 
+MqttMessageType = list[tuple[str, str]]
+AttributesType = list[tuple[str, Any]]
+StateDataType = list[tuple[MqttMessageType, str | None, AttributesType | None]]
```

Comment:
I think these should be moved to the top the file.
Also, if you only need them inside this module I think it would make sense to mark them as private.